app/perfsymboltable.cpp

```diff
@@ -379,25 +414,18 @@ Dwfl_Module *PerfSymbolTable::module(quint64 addr, const PerfElfMap::ElfInfo &el
     }
 
     Dwfl_Module *mod = dwfl_addrmodule(m_dwfl, addr);
+    quint64 baseAddr = elf.hasBaseAddr() ? elf.baseAddr : (elf.addr - elf.pgoff);
 
     if (!mod && elf.isValid()) {
         // check whether we queried for an address outside the elf range parsed
         // by dwfl. If that is the case, then we would invalidate the cache and
         // re-report the library again - essentially recreating the current state
         // for no gain, except wasting time
-        mod = dwfl_addrmodule(m_dwfl, elf.addr - elf.pgoff);
+        mod = dwfl_addrmodule(m_dwfl, baseAddr);
     }
 
-    if (mod) {
```

Comment:
I deleted this code because it was causing some tests to fail. I also tried replacing it with code that reads dwbias from dwfl_module_info but elfutils only provides that value if the file has DWARF debug info which caused some tests without DWARF to fail. I don't know how to reproduce the original problem ("many modules overlap ld.so"), and I'm not sure if it was a valid diagnosis.

Reply:
apparently some platforms still run into this @pcc - see e.g. https://github.com/KDAB/hotspot/issues/728#issuecomment-3113097727 and I think I also saw it on some other platform the other day but forgot again where it was and I'll keep my eyes open.

but I wonder: if I would want to re-instantiate the old code, what would I need to keep from your new code? such that I could see the test failures you saw to allow me to debug those?